src/multiselect.js

```diff
@@ -0,0 +1,206 @@
+   */
+  useDoubleClick_(on) {
+    if (!on) {
+      Blockly.Gesture.prototype.handleWsStart = this.origHandleWsStart_;
+      return;
+    }
+
+    Blockly.Gesture.prototype.handleWsStart = (function(func) {
+      if (func.isWrapped) {
+        return func;
+      } else {
+        const wrappedFunc = function(e, ws) {
+          func.call(this, e, ws);
+          if (this.targetBlock_ && e.buttons === 1 &&
+              !inMultipleSelectionMode) {
+            const preCondition = function(block) {
+              return !block.isInFlyout && block.isMovable() &&
+              block.workspace.options.collapse;
+            };
+            if (Blockly.selected && preCondition(Blockly.selected)) {
+              if (ws.doubleClickPid_) {
+                clearTimeout(ws.doubleClickPid_);
+                ws.doubleClickPid_ = undefined;
+                if (Blockly.selected.id === ws.doubleClickBlock_) {
+                  const state = !Blockly.selected.isCollapsed();
+                  const apply = function(block) {
+                    if (block && preCondition(block) &&
+                    !hasSelectedParent(block)) {
+                      block.setCollapsed(state);
+                    }
+                  };
+                  Blockly.Events.setGroup(true);
+                  if (Blockly.selected && !blockSelection.size) {
+                    apply(Blockly.selected);
+                  }
+                  blockSelection.forEach(function(id) {
+                    const block = ws.getBlockById(id);
+                    if (block) {
+                      apply(block);
+                    }
+                  });
+                  Blockly.Events.setGroup(false);
+                  return;
+                }
+              }
+              if (!ws.doubleClickPid_) {
+                ws.doubleClickBlock_ = Blockly.selected.id;
+                ws.doubleClickPid_ = setTimeout(function() {
+                  ws.doubleClickPid_ = undefined;
+                }, 500);
+              }
+            }
+          }
+        };
+        wrappedFunc.isWrapped = true;
+        return wrappedFunc;
+      }
+    })(Blockly.Gesture.prototype.handleWsStart);
+  }
+
+  /**
+   * Handle workspace events.
+   * @param {!Event} e Blockly event.
+   * @private
+   */
+  eventListener_(e) {
+    // on Block Selected
+    if (e.type === Blockly.Events.SELECTED) {
+      this.controls_.updateMultiselect();
+    }
+  }
+
+  /**
+   * Handle a key-down on the workspace.
+   * @param {KeyboardEvent} e The keyboard event.
+   * @private
+   */
+  onKeyDown_(e) {
+    if (e.keyCode === Blockly.utils.KeyCodes.SHIFT &&
+        !inMultipleSelectionMode) {
+      this.controls_.enableMultiselect();
+    }
+  }
+
+  /**
+   * Handle a key-up on the workspace.
+   * @param {KeyboardEvent} e The keyboard event.
+   * @private
+   */
+  onKeyUp_(e) {
+    if (e.keyCode === Blockly.utils.KeyCodes.SHIFT) {
+      this.controls_.disableMultiselect();
+    }
+  }
```

Comment:
Make sure that the following test works:

1. Press shift to enter multiselect mode
2. While holding shift, press alt+tab to switch to a different program
3. Use the mouse to reactivate the browser
4. Confirm that the system is no longer in multiselect mode

Reply:
Resolved by listening `window.blur`